src/psyclone/domain/gocean/transformations/gocean_opencl_trans.py

```diff
@@ -0,0 +1,157 @@
+            # Store the provided OpenCL options in the InvokeSchedule.
+            sched.set_opencl_options(options)
+
+        # The raised exceptions are converted to 'TransformationError's.
+        except (TypeError, AttributeError) as error:
+            six.raise_from(TransformationError(str(error)), error)
+
+        return sched, keep
+
+    def validate(self, sched, options=None):
+        '''
+        Checks that the supplied InvokeSchedule is valid and that an OpenCL
+        version of it can be generated.
+
+        :param sched: the Schedule to check.
+        :type sched: :py:class:`psyclone.psyGen.InvokeSchedule`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :raises TransformationError: if the InvokeSchedule is not for the \
+                                     GOcean1.0 API.
+        :raises NotImplementedError: if any of the kernels have arguments \
+                                     passed by value.
+        '''
+
+        if isinstance(sched, InvokeSchedule):
+            if not isinstance(sched, GOInvokeSchedule):
+                raise TransformationError(
+                    "OpenCL generation is currently only supported for the "
+                    "GOcean API but got an InvokeSchedule of type: '{0}'".
+                    format(type(sched)))
+        else:
+            raise TransformationError(
+                "Error in GOOpenCLTrans: the supplied node must be a (sub-"
+                "class of) InvokeSchedule but got {0}".format(type(sched)))
+
+        # Now we need to check the arguments of all the kernels
+        args = args_filter(sched.args, arg_types=["scalar"], is_literal=True)
+        for arg in args:
+            if arg.is_literal:
+                raise NotImplementedError(
+                    "Cannot generate OpenCL for Invokes that contain "
+                    "kernels with arguments passed by value")
+
+        # Check that we can construct the PSyIR and SymbolTable of each of
+        # the kernels in this Schedule. Also check that none of them access
+        # any form of global data (that is not a routine argument).
+        for kern in sched.kernels():
+            KernelTrans.validate(kern)
+            ksched = kern.get_kernel_schedule()
+            global_variables = ksched.symbol_table.global_symbols
+            if global_variables:
+                raise TransformationError(
+                    "The Symbol Table for kernel '{0}' contains the following "
+                    "symbols with 'global' scope: {1}. An OpenCL kernel cannot"
+                    " call other kernels and all of the data it accesses must "
+                    "be passed by argument. Use the KernelGlobalsToArguments "
+                    "transformation to convert such symbols to kernel "
+                    "arguments first.".
+                    format(kern.name, [sym.name for sym in global_variables]))
```

Comment:
Please add an `__all__` at the end of this file.

Reply:
Done